vertica_python/vertica/cursor.py

```diff
@@ -74,21 +78,21 @@ def execute(self, operation, parameters=None):
                 raise errors.Error("Argument 'parameters' must be dict or tuple")
 
         self.rowcount = 0
-        if self.last_execution:
-            # ToDo: can just empty the message buffer in connection if easy to do
-            self.connection.reset_connection()
+
         self.last_execution = operation
```

Comment:
I think last_execution was only used for the reset check, so you can probably get rid of it.